pandas/core/dtypes/common.py

```diff
@@ -1685,6 +1685,38 @@ def is_extension_type(arr):
     return False
 
 
+def is_extension_array_dtype(arr_or_dtype):
+    """Check if an object is a pandas extension array type
+
+    Parameters
+    ----------
+    arr_or_dtype : object
+
+    Returns
+    -------
+    bool
+
+    Notes
+    -----
+    This checks whether an object implements the pandas extension
+    array interface. In pandas, this includes:
+
+    * Categorical
+    * PeriodArray
+    * IntervalArray
+    * SparseArray
+
+    Third-party libraries may implement arrays or types satisfying
+    this interface as well.
+    """
+    from pandas.core.arrays import ExtensionArray
+
+    # we want to unpack series, anything else?
+    if isinstance(arr_or_dtype, ABCSeries):
+        arr_or_dtype = arr_or_dtype._values
+    return isinstance(arr_or_dtype, (ExtensionDtype, ExtensionArray))
```

Comment:
you need to call ``_get_dtype_type`` here, this can *only* have a result of ExtensionDtype and NOT ExtensionArray, which doesn't make any sense.

Reply:
> this can only have a result of ExtensionDtype and NOT ExtensionArray, which doesn't make any sense.

The result is just True or False. The argument can be either an array or dtype.

I'm not sure that `_get_dtype_or_type` does what we want here. That grabs `arr.dtype.type`, which is a scalar like `str` or `Interval` or `ipaddress.IPv4Address`. What would we do with that?

Reply:
this is not following what we do in all other cases that's my point. pls use _get_dtype_or_type

Reply:
> this is not following what we do in all other cases that's my point.

That won't work.

```
In [1]: import pandas as pd; import pandas_ip as ip

In [2]: arr = ip.IPAddress([1, 2, 3])

In [3]: pd.core.dtypes.common._get_dtype_type(arr)
Out[3]: pandas_ip.block.IPv4v6Base
```

`IPv4v6Base` isn't an instance of `ExtensionType`. It's the type scalars belong to.

```python
In [4]: isinstance(arr[0], ip.block.IPv4v6Base)
Out[4]: True

In [5]: issubclass(ip.block.IPv4v6Base, pd.core.dtypes.base.ExtensionDtype)
Out[5]: False
```

`_get_dtype_or_type` works for our extension types, since if we get a `CategoricalDtypeType` we can say "this is a categorical". But we can't do that for arbitrary 3rd party types.

Reply:
then ``_get_dtype_or_type`` needs adjustment. This is the point of compatibility, there shouldn't be the need to have special cases.

Reply:
But that would make `get_dtype_or_type` inconsistent, as it would no longer *always* return a dtype type, but in certain cases a dtype

Reply:
`_get_dtype_type` does exactly what it's supposed to do, `values.dtype.type`. But that's not useful here!

What's the issue with the function as defined? I need a way to tell if an array or dtype is an ExtensionArray or ExtensionDtype. Someday, when `Categorical`, `SparseArray`, `IntervalArray`, `PeriodArray`, `datetimetz`, etc are extension arrays then all the *special cases* currently in `_get_dtype_type` and friends can be removed, but we aren't there yet. We're doing things in small steps.